src/java.base/share/classes/java/nio/file/Path.java

```diff
@@ -248,6 +250,151 @@ public static Path of(URI uri) {
+     *
+     * @since 19
+     */
+    default String getExtension(String defaultExtension) {
+        Path fileName = getFileName();
+        if (fileName == null) {
+            return defaultExtension;
+        }
+
+        String fileNameString = fileName.toString();
+        int length = fileNameString.length();
+
+        // Indeterminate if fileNameString is too short
+        if (length > 2) {
+            int lastDotIndex = fileNameString.lastIndexOf('.');
+            // Indeterminate if no dot or found at last or only the first index
+            if (lastDotIndex > 0 && lastDotIndex < length - 1) {
+                return fileNameString.substring(lastDotIndex + 1);
+            }
+        }
+
+        return defaultExtension;
+    }
+
+    /**
+     * Returns whether the string representation of this path has an
+     * extension among the one or more extensions given as parameters.
+     * The extension of this path will be compared against the given
+     * parameters in the order in which they are supplied.
+     *
+     * @param   ext
+     *          the first extension for which to check
+     *
+     * @param   extensions
+     *          zero or more subsequent extensions for which to check
+     *
+     * @return  an {@code Optional} which either contains the found extension
+     *          or is empty if this path either does not have an extension or
+     *          its extension is not among those given
+     *
+     * @since 19
+     */
+    default Optional<String> hasExtension(String ext, String... extensions) {
+        String thisExtension = getExtension(null);
+        if (thisExtension == null)
+            return Optional.empty();
+
+        Objects.requireNonNull(ext);
+        if (ext.equals(thisExtension))
+            return Optional.of(ext);
+
+        for (String e : extensions) {
+            Objects.requireNonNull(e);
+            if (e.equals(thisExtension))
+                return Optional.of(e);
+        }
+
+        return Optional.empty();
+    }
+
+    /**
+     * Returns a new {@code Path} which is equal to this path aside from its
+     * extension which is replaced by the provided extension. If the provided
+     * extension is {@link String#isEmpty() empty}, then the current extension
+     * and the dot ('.') preceding it are removed. If this path does not have
+     * an extension then the provided extension is appended.
+     *
+     * @implNote
+     * This default implementation works with the string representation of this
+     * path. Paths which have a native representation should override this
+     * implementation in order to preserve that representation.
+     *
+     * @param  extension
+     *         the extension to replace this path's current extension;
+     *         leading and trailing space is ignored; may contain at most
+     *         one dot ('.') which if present must be the first character
+     *         that is not a space
```

Comment:
Is there a use case for supplying the new extension with leading or training space.
I think the API is clearer if the supplied extension is treated as a literal.

Reply:
No, I just thought it might be more flexible and forgiving.